### ISSUE TITLE
Reinterpreting a `TypedDict` with uniform value types as a `Mapping`

### ISSUE THREAD

**Issue by tmke8** [astral-sh/ty#4184]
### Summary

The following is sound, I think:

```python
from collections.abc import Mapping
from typing import TypedDict

class D(TypedDict):
    a: int
    b: int

def f(d: D) -> Mapping[str, int]:
    return d  # Return type does not match returned value: expected `Mapping[str, int]`, found `D` (invalid-return-type)
```

but ty gives an error on the last line: https://play.ty.dev/864abe11-7e72-40a0-b545-bf08c9fca851

To be fair, however, none of the other type checkers I've tried (pyright, mypy, pyrefly) accept this code either. So I'd understand if you don't want to support this.

### Version

playground, 7b025a54e

**Comment by AlexWaygood** [comment]
This isn't sound, because of the fact that TypedDicts are open by default! A subclass of `D` could add a non-`int` field, and you could pass in an "instance of that subclass" to the `d` parameter of `f`. If you add `closed=True` to the TypedDict definition, ty's error goes away.

We may be able to improve our diagnostic here to make the issue clearer.

**Comment by tmke8** [comment]
Oh, I see! Thanks, this works.

**Comment by AlexWaygood** [comment]
in the next version of ty, the diagnostic will look like this:

<img width="2340" height="452" alt="Image" src="https://github.com/user-attachments/assets/dca9cb42-fe02-4912-94b2-ff0adcf1db05" />

thanks for the report!
